processor/src/test/java/com/joom/lightsaber/processor/validation/ValidatorTest.kt

```diff
@@ -0,0 +1,34 @@
+package com.joom.lightsaber.processor.validation
+
+import com.joom.lightsaber.processor.ErrorReporter
+import com.joom.lightsaber.processor.end2end.End2EndTest
+import java.nio.file.Path
+import org.junit.Test
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+
+class ValidatorTest : End2EndTest(Path.of("src/test/resources/test_case_projects")) {
+
+  @Test
+  fun test_processed_without_any_problems() {
+    processProject("valid_configuration")
+  }
+
+  @Test
+  fun test_validation_fails_if_provided_by_is_used_without_inject() {
+    val reporterMock = mock<ErrorReporter>()
+    processProject(
+      "provided_by_without_inject",
+      reporterMock,
+      ignoreErrors = false
+    )
+
+    verify(reporterMock).reportError(
+      "Invalid configuration for dependency: " +
+          "test_case_projects.provided_by_without_inject.DependencyInterface. \n" +
+          "\ttest_case_projects.provided_by_without_inject.DependencyImplementation: " +
+          "Using @ProvidedBy without @Inject. You must either add at least one constructor " +
+          "with @Inject annotation or add @Provide method\n"
+    )
+  }
```

Comment:
I think that common usages should be built into the runner or provided as extensions so you can write test cases using a single command. For instance it could be:
```
  @Test
  fun test_processed_without_any_problems() {
    assertValidProject("valid_configuration", "<optional message>")
  }
``` 

or

```
    assertInvalidProject("provided_by_without_inject", "<optional message>")
```